core/src/main/scala/org/bitcoins/core/dlc/oracle/NonceSignaturePairDb.scala

```diff
@@ -0,0 +1,135 @@
+package org.bitcoins.core.dlc.oracle
+
+import org.bitcoins.core.protocol.dlc.models.{
+  EnumOracleOutcome,
+  NumericOracleOutcome
+}
+import org.bitcoins.core.protocol.tlv.{BaseOracleAnnouncement}
+import org.bitcoins.core.util.sorted.OrderedNonces
+import org.bitcoins.crypto.{FieldElement, SchnorrDigitalSignature, SchnorrNonce}
+
+/** Shim between v0 and v1 of the dlc spec and how we stored nonces
+  * this trait can be removed when we drop v0 of the spec
+  */
+sealed trait NonceSignaturePairDbShim {
+  def announcementId: Long
+  def nonce: SchnorrNonce
+  def signatureOpt: Option[SchnorrDigitalSignature]
+
+  def outcomeOpt: Option[String]
+
+  def attestationOpt: Option[FieldElement]
+
+}
+
+case class NonceSignaturePairDb(
+    announcementId: Long,
+    nonce: SchnorrNonce,
+    nonceProof: SchnorrDigitalSignature,
+    attestationOpt: Option[FieldElement],
+    outcomeOpt: Option[String])
+    extends NonceSignaturePairDbShim {
+//  require(attestationOpt.isDefined == outcomeOpt.isDefined,
```

Comment:
I don't think this can enabled. This works in the case of enum outcomes where we have a 1-1 relationship between a nonce and a DLC. 

However in the case of numeric outcomes that are `RemoteClaimed`, i don't think we can accurately reconstruct the attestation for the _specific_ nonce. The field element we recover from onchain is aggregated I believe.

Reply:
See this commit for some context

https://github.com/bitcoin-s/bitcoin-s/pull/4790/commits/252bfa44ed26cafc0b188e99ed872b7b192c2c08